### PR TITLE
create a new component HandleLink and use with TableCell

### PR DESCRIPTION
**Pull request by Francisco2002**
Creating a new component to allow opening a new tab.
Test on the /tree route

### CODE REVIEW

File: dashboard/src/components/HandleLink/HandleLink.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}) => {
+  const [linkTarget, setLinkTarget] = useState<string | undefined>(undefined);
+  const linkRef = useRef<HTMLAnchorElement>(null);
+
+  useEffect(() => {
+    if (linkTarget) {
+      setLinkTarget(undefined);
+      linkRef.current?.click();
+    }
+  }, [linkTarget]);
```
Comment (WilsonNet) [inline dashboard/src/components/HandleLink/HandleLink.tsx:17]: useEffect is a little too much for a simple link, if we want to move to server side rendering we will need to refactor to remove it

File: dashboard/src/components/HandleLink/HandleLink.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { LinkComponentProps } from 'node_modules/@tanstack/react-router/dist/esm/link';
```
Comment (WilsonNet) [inline dashboard/src/components/HandleLink/HandleLink.tsx:1]: `@types/react`  has `import { ComponentProps } from 'react';`
so you don't need to import stuff from node_modules 

also, every time you import a type use `import type`, we will be adding that eslint rule one day

**Review by WilsonNet — COMMENTED** [review]
The idea of making a TableCell with link component is nice, I liked it, but that `HandleLinkComponent` doesn't seem useful to me

File: dashboard/src/components/HandleLink/HandleLink.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import React, { PropsWithChildren, useEffect, useRef, useState } from 'react';
+import { Link } from '@tanstack/react-router';
+
+const HandleLink: React.FC<PropsWithChildren<LinkComponentProps<'a'>>> = ({
+  children,
+  ...props
+}) => {
+  const [linkTarget, setLinkTarget] = useState<string | undefined>(undefined);
+  const linkRef = useRef<HTMLAnchorElement>(null);
```
Comment (WilsonNet) [inline dashboard/src/components/HandleLink/HandleLink.tsx:10]: too many bloat for a simple link, it should be just use the Link component and that is it, no need for these hooks

File: dashboard/src/components/ui/table.tsx [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import * as React from 'react';
 
 import { cn } from '../../lib/utils';
+import HandleLink from '../HandleLink/HandleLink';
```
Comment (WilsonNet) [inline dashboard/src/components/ui/table.tsx:4]: I know we have some imports that are like `../` but please don't use that, import from `@/components`  
we already have a ticket for adding this eslint rule, the reason we had that is because our vite config was not supporting that in the beginning

File: dashboard/src/components/SideMenu/NavLink.tsx [hunk 5]
```diff
@@ -1,25 +1,37 @@
...
+const NavLink = ({ icon, idIntl, ...props }: INavLink): JSX.Element => {
+  return (
+    <NavigationMenuLink asChild>
+      <Link
+        className="flex items-center no-underline hover:text-sky-500"
+        {...props}
+      >
+        <span className="mr-3">{icon}</span>
+        <span className="text-center text-sm">
+          <FormattedMessage id={idIntl} />{' '}
```
[the remark this answers is not in the document]
Comment (Francisco2002) [reply dashboard/src/components/SideMenu/NavLink.tsx:30]: I think this was put in by the formatter

File: dashboard/src/components/ui/table.tsx [hunk 6]
```diff
@@ -93,6 +94,29 @@ const TableCell = React.forwardRef<
 ));
 TableCell.displayName = 'TableCell';
 
+type TableCellAttributes = React.TdHTMLAttributes<HTMLTableCellElement> & {
+  linkProps?: LinkProps
```
[the remark this answers is not in the document]
Comment (WilsonNet) [reply dashboard/src/components/ui/table.tsx:98]: My bad, it is just the name that doesn't include the `WithLink`, that is kind confusing, prefer big descriptive names, it is better to save lines of brain than lines of code

[the remark this answers is not in the document]
Comment (WilsonNet) [reply dashboard/src/components/ui/table.tsx:98]: And in this case `linkProps` is not optional, otherwise it would be just a `TableCell`

File: dashboard/src/components/ui/table.tsx [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import * as React from 'react';
 
 import { cn } from '../../lib/utils';
+import { Link, LinkProps } from '@tanstack/react-router';
```
Comment (WilsonNet) [inline dashboard/src/components/ui/table.tsx:4]: Had no idea they exported `LinkProps`, nice finding

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 8]
```diff
@@ -81,7 +83,11 @@ const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
       }
       key={item.idIntl}
     >
-      <NavLink icon={item.icon} idIntl={item.idIntl} onClick={navigate} />
+      <NavLink
+        to={navigate()}
```
Comment (MarceloRobert) [inline dashboard/src/components/SideMenu/SideMenu.tsx:87]: There seems to be an error here, can't assign any string to the property "to"

**Comment by Francisco2002** [comment]
1. Use Link Component of TanStack Library for give link behavior and can open new tabs in
2. Resolve issue #370

**Review by anajalvarenga — COMMENTED** [review]
If the commented codes are not used again, I recommend removing them

File: dashboard/src/components/Table/TreeTable.tsx [hunk 9]
```diff
@@ -57,57 +57,78 @@ const treeTableHeaders = headerLabelOrElement.map(item =>
...
+      target={linkTarget}
+      onContextMenu={(e) => {
+        e.preventDefault();
+        setLinkTarget("_blank");
+      }}
+    >
+      {children}
+    </Link>
+  );
+}*/
```
Comment (WilsonNet) [inline dashboard/src/components/Table/TreeTable.tsx:89]: Leftover?

File: dashboard/src/types/tree/Tree.tsx [hunk 10]
```diff
@@ -90,9 +90,29 @@ const origins = [
...
+  '/tree',
+  '/devices',
+  '/labs',
+  '/tree/$treeId',
+  '/tree/$treeId/test',
+  '/tree/$treeId/test/$testId',
+  './',
+  '../',
+  '/tree/$treeId/build/$buildId',
+] as const;
```
Comment (WilsonNet) [inline dashboard/src/types/tree/Tree.tsx:106]: These paths are only for the `SideMenu` so it should be in that file, and you can delete all the paths that are not a SideMenu lilnk

File: dashboard/src/types/tree/Tree.tsx [hunk 11]
```diff
@@ -90,9 +90,29 @@ const origins = [
...
+
 const DEFAULT_ORIGIN = 'maestro';
+const DEFAULT_PATH = '/tree';
```
Comment (WilsonNet) [inline dashboard/src/types/tree/Tree.tsx:109]: ditto about SideMenu

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 12]
```diff
@@ -64,14 +64,10 @@ const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
...
-    const origin = zOrigin.parse(unsafeOrigin);
+  const navigate = (): TPaths => {
     const finalPath = item.navigateTo;
-    const newPath = `${finalPath}?origin=${origin}`;
 
-    router.navigate({
-      to: newPath,
-    });
+    return zPath.parse(finalPath);
   };
```
Comment (WilsonNet) [inline dashboard/src/components/SideMenu/SideMenu.tsx:71]: this could be just an 

`const safeNavigateTo = zPath.parse(item.navigateTo)`

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 13]
```diff
@@ -81,7 +77,14 @@ const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
       }
       key={item.idIntl}
     >
-      <NavLink icon={item.icon} idIntl={item.idIntl} onClick={navigate} />
+      <NavLink
+        to={navigate()}
```
[the remark this answers is not in the document]
Comment (WilsonNet) [reply dashboard/src/components/SideMenu/SideMenu.tsx:81]: https://github.com/kernelci/dashboard/pull/359/files#r1795950202

File: dashboard/src/components/Summary/Summary.tsx [hunk 14]
```diff
@@ -71,24 +73,60 @@ export const SummaryItem = ({
...
+            } else {
+              configs[arch.text] = true;
+            }
+
+            return {
+              ...previousParams,
+              diffFilter: newFilter,
+            };
+          },
+        }}
```
Comment (WilsonNet) [inline dashboard/src/components/Summary/Summary.tsx:128]: ditto about logic reuse

File: dashboard/src/components/ui/table.tsx [hunk 15]
```diff
@@ -66,6 +67,34 @@ const TableRow = React.forwardRef<
...
+    )}
+    {...props}
+  >
+    <Link
+      className={cn('flex flex-1')}
+      {...linkProps}
+    >
+        {children}
+      </Link>
+  </tr>
```
Comment (WilsonNet) [inline dashboard/src/components/ui/table.tsx:94]: The child of a `tr` should be a `td`, <a> is not valid html

Comment (WilsonNet) [reply dashboard/src/components/ui/table.tsx:94]: image

its creating this console warning (yes we have some other warnings for invalid html that you dind't create, we will need to fix those another day also)

File: dashboard/src/pages/TreeDetails/Tabs/TestCards.tsx [hunk 16]
```diff
@@ -37,22 +37,44 @@ const ConfigsList = ({
...
+                  if (configs[configName]) {
+                    delete configs[configName];
+                  } else {
+                    configs[configName] = true;
+                  }
+
+                  return {
+                    ...previousParams,
+                    diffFilter: newFilter,
+                  };
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeDetails/Tabs/TestCards.tsx:58]: ditto about the logic being reused

**Review by WilsonNet — COMMENTED** [review]
nice job, it is working great but the code can be improved

File: dashboard/src/components/Accordion/Accordion.tsx [hunk 17]
```diff
@@ -276,11 +273,11 @@ const TestTableRow = ({ test, onClick }: ITestTableRow): JSX.Element => {
           showLabelTZ={true}
         />
       </TableCell>
-      <TableCell>{test.duration ?? '-'}</TableCell>
-      <TableCell>
+      <TableCell>{test.duration || '-'}</TableCell>
```
Comment (mari1912) [inline dashboard/src/components/Accordion/Accordion.tsx:276]: why did you change this?

File: dashboard/src/components/SideMenu/NavLink.tsx [hunk 18]
```diff
@@ -1,25 +1,40 @@
...
+
+type INavLink = LinkProps & {
+  idIntl: MessagesKey;
+  icon: JSX.Element;
+  href?: string;
+  as?: string;
+};
+
+const NavLink = ({ icon, idIntl, as, ...props }: INavLink): JSX.Element => {
+  const LinkElement = as || Link;
```
Comment (mari1912) [inline dashboard/src/components/SideMenu/NavLink.tsx:23]: nit: same thing, is more common to use `??` instead of `||`

Comment (mari1912) [reply dashboard/src/components/SideMenu/NavLink.tsx:23]: Also, why are you using this LinkElement? IMO it would be better to add a link and make it disabled or doing nothing when clicked, but I don't have the entire context

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 19]
```diff
@@ -53,7 +53,7 @@ type SideMenuItemProps = {
 const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
   const { pathname } = useLocation();
 
-  const router = useRouter();
+  //const router = useRouter();
```
Comment (mari1912) [inline dashboard/src/components/SideMenu/SideMenu.tsx:56]: nit: if you are not using it, you should remove

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 20]
```diff
@@ -81,7 +77,14 @@ const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
...
       key={item.idIntl}
     >
-      <NavLink icon={item.icon} idIntl={item.idIntl} onClick={navigate} />
+      <NavLink
+        to={navigate()}
+        search={{
+          origin: zOrigin.parse(unsafeOrigin),
+        }}
+        icon={item.icon}
+        idIntl={item.idIntl} /*onClick={navigate}*/
```
Comment (mari1912) [inline dashboard/src/components/SideMenu/SideMenu.tsx:86]: ditto

File: dashboard/src/components/Summary/Summary.tsx [hunk 21]
```diff
@@ -71,24 +73,60 @@ export const SummaryItem = ({
...
+            delete configs[compiler];
+          } else {
+            configs[compiler] = true;
+          }
+
+          return {
+            ...previousParams,
+            diffFilter: newFilter,
+          };
+        }}
```
[the remark this answers is not in the document]
Comment (mari1912) [reply dashboard/src/components/Summary/Summary.tsx:98]: yes, try to make small components and functions to improve readability

File: dashboard/src/components/Summary/Summary.tsx [hunk 22]
```diff
@@ -71,24 +73,60 @@ export const SummaryItem = ({
...
-  onClickCompiler,
   leftIcon,
 }: ISummaryItem): JSX.Element => {
   const compilersElement = useMemo(() => {
     return compilers?.map(compiler => (
-      <button
+      <Link
+        search={previousParams => {
+          const { diffFilter: currentDiffFilter } = previousParams;
+          const newFilter = structuredClone(currentDiffFilter) || {};
```
Comment (mari1912) [inline dashboard/src/components/Summary/Summary.tsx:83]: ditto

File: dashboard/src/components/Table/BaseTable.tsx [hunk 23]
```diff
@@ -69,13 +75,19 @@ const BaseTable = ({
...
   return (
     <div className="h-full">
       <DumbBaseTable className={className}>
-        <DumbTableHeader>
+        <DumbTableHeader gridClassName={gridClassName}>
           {headers.map(column => (
-            <TableHead className="border-b text-black" key={column.key}>
+            <TableHead
+              className={gridClassName ? 'flex items-center' : ''}
+              /*className={`flex flex-col justify-center w-[${fieldSizes[index]}%]`}*/ key={
```
Comment (mari1912) [inline dashboard/src/components/Table/BaseTable.tsx:87]: ditto

File: dashboard/src/components/Table/BootsTable.tsx [hunk 24]
```diff
@@ -124,24 +110,35 @@ const BootsTable = ({ treeId, testHistory }: ITestsTable): JSX.Element => {
...
+          params: {
+            treeId,
+            testId: test.id,
+          },
+          search: s => s,
+        }}
+      >
+        <TableCell className="">{test.path}</TableCell>
+        <TableCell className="">{test.status}</TableCell>
+        <TableCell className="">
```
Comment (mari1912) [inline dashboard/src/components/Table/BootsTable.tsx:127]: remove the className if you are not using it

File: dashboard/src/components/Table/BootsTable.tsx [hunk 25]
```diff
@@ -124,24 +110,35 @@ const BootsTable = ({ treeId, testHistory }: ITestsTable): JSX.Element => {
...
           <TooltipDateTime
             dateTime={test.startTime}
             lineBreak={true}
             showLabelTime={true}
             showLabelTZ={true}
           />
         </TableCell>
-        <TableCell>{test.duration ?? '-'}</TableCell>
-        <TableCell>
+        <TableCell className="">{test.duration || '-'}</TableCell>
```
Comment (mari1912) [inline dashboard/src/components/Table/BootsTable.tsx:135]: ditto

File: dashboard/src/components/Table/BootsTable.tsx [hunk 26]
```diff
@@ -124,24 +110,35 @@ const BootsTable = ({ treeId, testHistory }: ITestsTable): JSX.Element => {
...
     return filteredData?.slice(startIndex, endIndex).map(test => (
-      <TableRow onClick={() => onClickName(test.id)} key={test.id}>
-        <TableCell>{test.path}</TableCell>
-        <TableCell>{test.status}</TableCell>
-        <TableCell>
+      <TableRowWithLink
+        key={test.id}
+        linkProps={{
+          to: '/tree/$treeId/test/$testId',
+          className: 'grid grid-cols-[25%_15%_15%_25%_15%]',
```
Comment (mari1912) [inline dashboard/src/components/Table/BootsTable.tsx:117]: you can remove this to a const and reuse it in here and in the other tableRowWithLink, it will make it easier to change in the future

File: dashboard/src/components/Table/TreeTable.tsx [hunk 27]
```diff
@@ -57,57 +57,78 @@ const treeTableHeaders = headerLabelOrElement.map(item =>
...
-          treeInfo: {
-            gitUrl: row.url,
-            gitBranch: row.branch,
-            treeName: row.tree_name ?? undefined,
-            commitName: row.commitName,
-            headCommitHash: row.id,
-          },
-        },
-      });
+  // const navigate = useNavigate({ from: '/tree' });
```
Comment (mari1912) [inline dashboard/src/components/Table/TreeTable.tsx:95]: ditto

File: dashboard/src/components/Table/TreeTable.tsx [hunk 28]
```diff
@@ -57,57 +57,78 @@ const treeTableHeaders = headerLabelOrElement.map(item =>
...
-      row.tree_name,
-      row.commitName,
-      origin,
-    ],
-  );
+    search: {
+      tableFilter: {
+        bootsTable: possibleTestsTableFilter[0],
+        buildsTable: possibleBuildsTableFilter[2],
+        testsTable: possibleTestsTableFilter[0],
```
Comment (mari1912) [inline dashboard/src/components/Table/TreeTable.tsx:111]: is this correct? to have the same values in boots and tests?

File: dashboard/src/components/Table/TreeTable.tsx [hunk 29]
```diff
@@ -57,57 +57,78 @@ const treeTableHeaders = headerLabelOrElement.map(item =>
...
+      },
+      origin: origin,
+      currentTreeDetailsTab: zPossibleValidator.parse(target),
+      diffFilter: {},
+      treeInfo: {
+        gitUrl: row.url,
+        gitBranch: row.branch,
+        treeName: row.tree_name ?? undefined,
```
Comment (mari1912) [inline dashboard/src/components/Table/TreeTable.tsx:119]: i think that you don't need the `??` here, considering it will already be undefined or null if there is no tree_name

File: dashboard/src/components/Table/TreeTable.tsx [hunk 30]
```diff
@@ -57,57 +57,78 @@ const treeTableHeaders = headerLabelOrElement.map(item =>
...
+      },
+    },
+  });
 
   return (
     <TableRow>
-      <TableCell
-        onClick={navigateToTreeDetailPage}
+      <TableCellWithLink
+        /*onClick={navigateToTreeDetailPage}*/
```
Comment (mari1912) [inline dashboard/src/components/Table/TreeTable.tsx:129]: remove all the onClick comments

File: dashboard/src/components/Table/TreeTable.tsx (hunk 28 above)
Comment (Francisco2002) [reply dashboard/src/components/Table/TreeTable.tsx:111]: This code was already like this when I got it. I reported this.

File: dashboard/src/components/Table/TreeTable.tsx (hunk 29 above)
Comment (Francisco2002) [reply dashboard/src/components/Table/TreeTable.tsx:119]: ditto

File: dashboard/src/components/SideMenu/NavLink.tsx (hunk 18 above)
Comment (Francisco2002) [reply dashboard/src/components/SideMenu/NavLink.tsx:23]: If you click in Documentation option in sidemenu, you will see a redirect for \tree route. This isn't the expected behavior. This should redirect to an external page. By my tests, the Link component has this behavior. Because this, I use a dinamic link wrapper (Tanstack Link or HTML link).

File: dashboard/src/components/Accordion/Accordion.tsx [hunk 31]
```diff
@@ -223,63 +228,51 @@ const AccordionTestsTrigger = ({
...
-const TestTableRow = ({ test, onClick }: ITestTableRow): JSX.Element => {
-  const onClickHandle = useCallback(() => onClick(test.id), [onClick, test.id]);
+const TestTableRow = ({ test }: ITestTableRow): JSX.Element => {
+  const linkProps: LinkProps = {
+    to: '/tree/$treeId/test/$testId',
+    params: {
+      testId: test.id,
+    },
+    search: s => s,
+  };
```
Comment (WilsonNet) [inline dashboard/src/components/Accordion/Accordion.tsx:253]: `useMemo`, objects will always create a new reference, `console.log({} === {})` is false

File: dashboard/src/utils/filters.ts [hunk 32]
```diff
@@ -45,3 +51,25 @@ export const getTargetFilter = (
 
   return acc;
 };
+
+export const useDiffFilterParams = (
```
Comment (WilsonNet) [inline dashboard/src/utils/filters.ts:55]: this will work only in the TreeDetailsPage
so I woudl move it to  `pages/TreeDetails/utils/treeDetailsFilters.ts`

File: dashboard/src/pages/TreeDetails/Tabs/TestCards.tsx [hunk 33]
```diff
@@ -19,11 +23,37 @@ import { groupStatus } from '@/utils/status';
 import ColoredCircle from '@/components/ColoredCircle/ColoredCircle';
 import { TIssue } from '@/types/general';
 import { NoIssueFound } from '@/components/Issue/IssueSection';
+import { useDiffFilterParams } from '@/utils/filters';
 
 interface IConfigList extends Pick<TTreeTestsData, 'configStatusCounts'> {
   title: IBaseCard['title'];
 }
 
+const LinkWrapper = ({
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeDetails/Tabs/TestCards.tsx:32]: Since these link wrappers are completely equal, you can create only one in something like `Pages/TreeDetails/TreeDetailsFilterLink.tsx` with the name `TreeDetailsFilterLink` and import them. 

If there is a slightly difference between them is okay to have more than one and keep them inside the component, but they seem to be completely equal, so you can just reuse a component

File: dashboard/src/components/Summary/Summary.tsx [hunk 34]
```diff
@@ -67,28 +73,57 @@ const Summary = ({
   );
 };
 
+const LinkWrapper = ({
```
Comment (WilsonNet) [inline dashboard/src/components/Summary/Summary.tsx:76]: ditto about link wrapper

File: dashboard/src/pages/TreeDetails/Tabs/Build/BuildTab.tsx [hunk 35]
```diff
@@ -91,9 +93,33 @@ const StatusCard = ({
 
 const MemoizedStatusCard = memo(StatusCard);
 
+const LinkWrapper = ({
```
Comment (WilsonNet) [inline dashboard/src/pages/TreeDetails/Tabs/Build/BuildTab.tsx:96]: ditto about link wrapper

File: dashboard/src/components/Accordion/BuildAccordionContent.tsx [hunk 36]
```diff
@@ -1,10 +1,10 @@
 import { MdFolderOpen } from 'react-icons/md';
 
-import { useMemo, useCallback } from 'react';
+import { useCallback, useMemo } from 'react';
 
 import { FormattedMessage } from 'react-intl';
 
-import { useNavigate, useParams } from '@tanstack/react-router';
+import { useParams, useNavigate } from '@tanstack/react-router';
```
Comment (WilsonNet) [inline dashboard/src/components/Accordion/BuildAccordionContent.tsx:7]: is there ar eason for the changes in this file? it seems a little like bit like noise

File: dashboard/src/components/SideMenu/SendFeedback.tsx [hunk 37]
```diff
@@ -68,14 +74,14 @@ const SendFeedback = (
...
             <AlertDialogAction asChild>
-              <a onClick={onGitHubClick}>
+              <Link onClick={onGitHubClick}>
                 <FormattedMessage id="global.github" />
-              </a>
+              </Link>
             </AlertDialogAction>
             <AlertDialogAction asChild>
-              <a onClick={onEmailClick}>
+              <Link onClick={onEmailClick}>
```
Comment (WilsonNet) [inline dashboard/src/components/SideMenu/SendFeedback.tsx:82]: ditto about link, these are external links, they don't need a Link component nor an onclick

File: dashboard/src/components/SideMenu/SendFeedback.tsx [hunk 38]
```diff
@@ -68,14 +74,14 @@ const SendFeedback = (
           </AlertDialogHeader>
           <AlertDialogFooter>
             <AlertDialogAction asChild>
-              <a onClick={onGitHubClick}>
+              <Link onClick={onGitHubClick}>
```
[the remark this answers is not in the document]
Comment (WilsonNet) [reply dashboard/src/components/SideMenu/SendFeedback.tsx:77]: instead of `getGithubBody` to be afunction, it will be just an useMemo based on that state

[the remark this answers is not in the document]
Comment (WilsonNet) [reply dashboard/src/components/SideMenu/SendFeedback.tsx:77]: also, instead of window I think we should probably use `TanstackRouter`, it probably has a way to get the path 
take a look at

https://tanstack.com/router/latest/docs/framework/react/api/router/useLocationHook

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 39]
```diff
@@ -64,15 +64,12 @@ const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
...
   const isCurrentPath = pathname.startsWith(item.navigateTo);
 
-  const navigate = (): void => {
-    const origin = zOrigin.parse(unsafeOrigin);
-    const finalPath = item.navigateTo;
-    const newPath = `${finalPath}?origin=${origin}`;
+  const paths = ['/tree', '/devices', '/labs'] as const;
+
+  const zPathEnum = z.enum(paths);
+  const zPath = zPathEnum.catch('/tree');
```
Comment (WilsonNet) [inline dashboard/src/components/SideMenu/SideMenu.tsx:70]: nit: I think `zPathValidator` is a better name

File: dashboard/src/components/SideMenu/SideMenu.tsx [hunk 40]
```diff
@@ -64,15 +64,12 @@ const SideMenuItem = ({ item }: SideMenuItemProps): JSX.Element => {
...
 
-    router.navigate({
-      to: newPath,
-    });
-  };
+  const safeNavigateTo = zPath.parse(item.navigateTo);
```
Comment (WilsonNet) [inline dashboard/src/components/SideMenu/SideMenu.tsx:72]: Actually since item is defined in code level, it doesn't even need zod (it doesn't word to leave it here), this could be just the item prop type definition

File: dashboard/src/components/Table/TreeTable.tsx (hunk 28 above)
Comment (WilsonNet) [reply dashboard/src/components/Table/TreeTable.tsx:78]: this code is wrong, can you put a TODO to fix it after?, It is working but it is wrong

File: dashboard/src/components/Table/TreeTable.tsx [hunk 41]
```diff
@@ -61,53 +61,40 @@ const TreeTableRow = (row: TreeTableBody): JSX.Element => {
...
-            gitBranch: row.branch,
-            treeName: row.tree_name ?? undefined,
-            commitName: row.commitName,
-            headCommitHash: row.id,
-          },
-        },
-      });
+  const linkProps = (
+    target: string,
+  ): LinkProps & { style: React.CSSProperties } => ({
```
Comment (WilsonNet) [inline dashboard/src/components/Table/TreeTable.tsx:66]: ditto about `useMemo`

File: dashboard/src/components/Summary/Summary.tsx [hunk 42]
```diff
@@ -67,28 +73,57 @@ const Summary = ({
...
+      <LinkWrapper
         key={compiler}
-        className="line-clamp-1"
-        onClick={() => onClickCompiler?.(compiler)}
-      >
-        {compiler}
-      </button>
+        filterSection="compilers"
+        filterValue={compiler}
+      />
```
Comment (anajalvarenga) [inline dashboard/src/components/Summary/Summary.tsx:113]: The component's name is LinkWrapper but in this case, it doesn't wrap anything.

In my opinion, it would be more interesting to use the filterValue as a children, or change the component name

File: dashboard/src/components/Accordion/BuildAccordionContent.tsx (hunk 36 above)
Comment (MarceloRobert) [reply dashboard/src/components/Accordion/BuildAccordionContent.tsx:7]: I agree, the linter doesn't seem to bother with the order of this import

**Review by MarceloRobert — CHANGES_REQUESTED** [review]
There is a bug where if you are on a Tree Details page and click on the "Send Us Feedback" and then click on either GitHub or Email buttons, the current page will lose its parameters and an error will be shown instead of the right page. I believe the issue is with using Link instead of a on the SendFeedback.tsx file

File: dashboard/src/components/SideMenu/SendFeedback.tsx (hunk 38 above)
[the remark this answers is not in the document]
Comment (Francisco2002) [reply dashboard/src/components/SideMenu/SendFeedback.tsx:77]: Using useLocation Hook from TanstackRouter, we access only the pathname. The address BASE_URL/tree?origin return /tree, but we need the complete url. Why would I use this hook if the window.location.href return the complete url?

**Review by MarceloRobert — APPROVED** [review]
Working great!

**Review by anajalvarenga — COMMENTED** [review]
Working great! Just take a look at the commits to make them atomic

**Review by anajalvarenga — APPROVED** [review]
Nice job!
